File: Example/UnidaysSampleAppObjc/UnidaysSampleAppObjc/ViewController.m
//
//  ViewController.m
//  UnidaysSampleAppObjc
//
//  Created by Adam Mitchell on 29/01/2018.
//  Copyright © 2018 MyUNiDAYS. All rights reserved.
//

#import "ViewController.h"
@import UnidaysVerificationSDK;

@interface ViewController ()

@end

@implementation ViewController

- (IBAction)getCode:(id)sender {
    NSError *error = nil;
    UnidaysConfig *config = [[UnidaysConfig alloc] initWithScheme:@"sampleapp" customerSubdomain:self.subdomainTextField.text];
    [[UnidaysSDK sharedInstance] setupWithSettings:config error: &error];
    if (!error) {
        UnidaysChannel channel = self.channelSegmentedControl.selectedSegmentIndex == 1 ? UnidaysChannelOnline : UnidaysChannelInstore;
        self.barcodeImageView.image = nil;
        [[UnidaysSDK sharedInstance] getCodeWithChannel:channel   withSuccessHandler:^(id<CodeResultProtocol> _Nonnull result) {
            self.codeTextField.text = result.code;
            if (result.imageUrl != nil) {
                [self downloadImage:result.imageUrl];
            }
        } withErrorHandler:^(NSError * _Nonnull error) {
            self.errorLabel.text = error.localizedDescription;
        }];
    }
}

- (void)downloadImage:(NSURL*)withUrl {
    dispatch_async(dispatch_get_global_queue(0,0), ^{
        NSData * data = [[NSData alloc] initWithContentsOfURL: withUrl];
        if ( data == nil )
            return;
        dispatch_async(dispatch_get_main_queue(), ^{
            self.barcodeImageView.image = [UIImage imageWithData: data];
        });
    });
}

@end

File: Example/UnidaysSampleApp/UnidaysSampleApp/ViewController.swift
//
//  ViewController.swift
//  ROFLApp
//
//  Created by Adam Mitchell on 28/11/2017.
//  Copyright © 2017 MyUNiDAYS. All rights reserved.
//

import UIKit
import UnidaysVerificationSDK

class ViewController: UIViewController {
    
    let scheme = "sampleapp"
    
    @IBOutlet weak var codeTextField: UITextField!
    @IBOutlet weak var errorLabel: UILabel!
    @IBOutlet weak var subdomainTextField: UITextField!
    @IBOutlet weak var channelSegmentedControl: UISegmentedControl!
    @IBOutlet weak var codeImageView: UIImageView!
    
    @IBAction func onGetACodeTapped(_ sender: Any) {
        
        let sdk = UnidaysSDK.sharedInstance
        let settings = UnidaysConfig(scheme: scheme, customerSubdomain: self.subdomainTextField.text!)
        try! sdk.setup(settings: settings)
        let channels = [UnidaysChannel.Online, UnidaysChannel.Instore]
        let channel = channels[channelSegmentedControl.selectedSegmentIndex]
        sdk.getCode(channel: channel, success: { (response) in
            self.codeTextField.text = response.code
            if let imageUrl = response.imageUrl as URL? {
                self.download(url: imageUrl)
            } else {
                self.codeImageView.image = nil
            }
            self.errorLabel.text = nil
        }, error: { (error) in
            self.errorLabel.text = error.localizedDescription
            self.codeTextField.text = nil
            self.codeImageView.image = nil
        })
    }
    
    func download(url: URL) {
        let session = URLSession(configuration: .default)
        //creating a dataTask
        let getImageFromUrl = session.dataTask(with: url, completionHandler: { (data, response, error) in
            
            //if there is any error
            if let e = error {
                //displaying the message
                print("Error Occurred: \(e)")
                
            } else {
                //in case of now error, checking wheather the response is nil or not
                if (response as? HTTPURLResponse) != nil {
                    
                    //checking if the response contains an image
                    if let imageData = data {
                        
                        //getting the image
                        let image = UIImage(data: imageData)
                        
                        //displaying the image
                        DispatchQueue.main.async {
                            self.codeImageView.image = image
                        }
                    } else {
                        print("Image file is currupted")
                    }
                } else {
                    print("No response from server")
                }
            }
        })
        
        //starting the download task
        getImageFromUrl.resume()
    }
}
